verticapy/utilities.py

```diff
@@ -1831,12 +1896,16 @@ def read_csv(
             input_relation = "v_temp_schema.{}".format(quote_ident(table_name))
         file_header = []
         path_first_file_in_folder = path
-        if multiple_files:
+        if multiple_files and ingest_local:
             path_first_file_in_folder = get_first_file(path, "csv")
-        if not (header_names) and not (dtype) and (compression == "UNCOMPRESSED"):
+        if not (header_names) and not (dtype) and (compression == "UNCOMPRESSED") and ingest_local:
             if not (path_first_file_in_folder):
                 raise ParameterError("No CSV file detected in the folder.")
             file_header = get_header_name_csv(path_first_file_in_folder, sep)
+        elif not (header_names) and not (dtype) and (compression != "UNCOMPRESSED"):
+            raise ParameterError("The input file is compressed and parameters 'dtypes' and 'header_names' are not defined. It is impossible to read the file's header.")
```

Comment:
a) "dtype" instead of "dtypes"
b) Instead of "impossible", maybe switch to : "It is not possible to read..." 
OR 
"The headers of the the compressed cannot be read. Please specify dtype or header_names."

Reply:
Done;